ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js

```diff
@@ -0,0 +1,183 @@
+                    .sort()
+                    .reduce((accumulator, currentValue) => {
+                        accumulator.availabletypes.push({ metacardType: currentValue });
+                        return accumulator;
+                    }, { availabletypes: [] });
+                this.model.set('availableTypes', allTypes);
+                this.$el.addClass('is-selecting-available-types');
+            },
+            handleAvailableTypes() {
+
+                const availableTypes = this.model.get('availableTypes');
+
+                if(this.isSingleAvailableType()) {
+                    this.filterInjected = true;
+                    this.model.set('selectedAvailableType', this.model.get('availableTypes').availabletypes[0].metacardType);
+                } else if(this.isMultipleAvailableTypes()) {
+                    this.filterInjected = false;
+                    this.$el.addClass('is-selecting-available-types');
+                } else {
+                    this.handleSystemTypes();
+                }
+            },
+            handleSelectedAvailableType() {
+                this.$el.removeClass('is-selecting-available-types');
+
+                const metacardDefinition = metacardDefinitions.metacardDefinitions[this.model.get('selectedAvailableType')];
+
+                const nonInjectedAttributeNames = Object.keys(metacardDefinition)
+                    .filter((attributeName) => !metacardDefinitions.isHiddenType(attributeName))
+                    .filter((attributeName) => !metacardDefinition[attributeName].readOnly)
+                    .filter((attributeName) => !this.filterInjected || !metacardDefinition[attributeName].isInjected)
+                    .filter((attributeName) => attributeName !== "id");
+
+                const propertyCollection = {
+                    'metacard-type': this.model.get('selectedAvailableType')
+                };
+
+                nonInjectedAttributeNames.forEach((attribute) => {
+                    if (metacardDefinition[attribute].multivalued) {
+                      propertyCollection[attribute] = [];
+                    } else if (metacardDefinitions.enums[attribute]) {
+                      propertyCollection[attribute] = metacardDefinitions.enums[attribute][0];
+                    } else {
+                      propertyCollection[attribute] = "";
+                    }
+                });
+
+                this.model.set('metacard', propertyCollection);
+                this.$el.addClass('is-building');
+
+            },
+            handleMetacard() {
+                this.builderProperties.show(PropertyCollectionView.generatePropertyCollectionView([this.model.get('metacard')]));
+                this.builderProperties.currentView.$el.addClass("is-list");
+
+            },
+            onBeforeShow() {
+
+                    this.builderAvailableType.show(DropdownView.createSimpleDropdown({
```

Comment:
We should be able to eliminate this component by utilizing the property component.  

Here is an example: https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/theme-settings/theme-settings.view.js#L136